containerd-shim/main.go

```diff
@@ -106,7 +106,7 @@ func start(log *os.File) error {
 		case s := <-signals:
 			switch s {
 			case syscall.SIGCHLD:
-				exits, _ := osutils.Reap()
+				exits, _ := osutils.Reap(true)
```

Comment:
Ok, so this ought to be false indeed, or it would block the shim from handling other requests properly.

We get `0` returned by `wait4` because of the initial `runc` process running in detached mode:
  - we run `runc` and reap for it
  - receive the `SIGCHLD`, but runc was already reaped and the new process is alive, so we get `0` telling us that the child is not dead yet